### PR TITLE
Improvements over #1120

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: #814 
This PR is a proposed improvement over #1120 by @vishusidana95, which I felt, wasn't along the lines of SymPy's method of execution. As always, please review and suggest improvements.

### CODE REVIEW

File: symengine/functions.h [hunk 1]
```diff
@@ -529,6 +529,7 @@ class Dirichlet_eta : public OneArgFunction
     RCP<const Basic> rewrite_as_zeta() const;
     //! \return Canonicalized zeta
     virtual RCP<const Basic> create(const RCP<const Basic> &arg) const;
+    virtual RCP<const Basic> create(const vec_basic &arg) const;
```
Comment (isuruf) [inline symengine/functions.h]: This is not needed here as it is defined in OneArgFunction. You can make it visible by adding `using OneArgFunction::create`

File: symengine/derivative.cpp [hunk 2]
```diff
@@ -324,6 +323,12 @@ class DiffImplementation
         return fdiff(self, x);
     }
 
+    static RCP<const Basic> diff(const Dirichlet_eta &self,
```
Comment (isuruf) [inline symengine/derivative.cpp]: Is this necessary?

Comment (ShikharJ) [inline symengine/derivative.cpp]: @isuruf `Dirichlet_eta` is a `OneArgFunction` and automatic differentiation via the `fdiff` routine is enabled by default only for `TwoArgFunction`.

**Comment by ShikharJ** [comment]
ping @isuruf.

**Comment by ShikharJ** [comment]
ping @isuruf

File: symengine/derivative.cpp (hunk 2 above)
Comment (isuruf) [inline symengine/derivative.cpp]: Can you try making this `OneArgFunction` instead of `Dirichlet_eta` then? If that doesn't work, I'll merge this

Comment (ShikharJ) [inline symengine/derivative.cpp]: @isuruf Done.

**Comment by ShikharJ** [comment]
ping @isuruf.

**Pull request merged**